GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -341,6 +397,15 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
                     this.HasFailures |= !this.UpdateRefSpec(this.Tracer, this.Enlistment, branchOrCommit, refs);
                 }
             }
+
+            if (!this.HasFailures)
+            {
+                this.SavePrefetchArgs(commitToFetch, hydrateFilesAfterDownload);
+            }
+            else
+            {
+                this.SavePrefetchArgs(null, false);

```

Comment:
Is there a downside to leaving the old prefetch args (from the last successful prefetch) on disk if this prefetch fails?

That would make the failure case consistent with the scenario where FastFetch crashes or is killed.